lighthouse-core/gather/gatherers/image-usage.js

```diff
@@ -167,27 +167,24 @@ class ImageUsage extends Gatherer {
       return (${collectImageElementInfo.toString()})();
     })()`;
 
-    /** @type {Promise<Array<ImageSizeInfo>>} */
-    const evaluatePromise = driver.evaluateAsync(expression);
-    return evaluatePromise.then(elements => {
-      return elements.reduce((promise, element) => {
-        return promise.then(collector => {
-          // Images within `picture` behave strangely and natural size information isn't accurate,
-          // CSS images have no natural size information at all.
-          // Try to get the actual size if we can.
-          const elementPromise = (element.isPicture || element.isCss) && element.networkRecord ?
-              this.fetchElementWithSizeInformation(driver, element) :
-              Promise.resolve(element);
-
-          return elementPromise.then(element => {
-            // link up the image with its network record
-            element.networkRecord = indexedNetworkRecords[element.src];
-            collector.push(/** @type {LH.Artifacts.SingleImageUsage} */ (element));
-            return collector;
-          });
-        });
-      }, Promise.resolve(/** @type {LH.Artifacts['ImageUsage']} */ ([])));
-    });
+    /** @type {Array<ImageSizeInfo>} */
+    const elements = await driver.evaluateAsync(expression);
+
+    /** @type {LH.Artifacts['ImageUsage']} */
+    const imageUsage = [];
+    for (let element of elements) {
+      // Images within `picture` behave strangely and natural size information isn't accurate,
+      // CSS images have no natural size information at all.
+      // Try to get the actual size if we can.
+      if (element.isPicture || element.isCss) {
```

Comment:
should we maintain the original concern and only fetch when we can find the network record? `fetchElementWithSize` is the expensive path

Reply:
> should we maintain the original concern and only fetch when we can find the network record? 

Only `uses-responsive-images` seems to use `naturalWidth`/`naturalHeight` from css/`<picture>` images, and it requires a networkRecord to work, so makes sense

Reply:
> when we can find the network record?

...actually this conversation clearly shows that it should have been an optional property...